packages/editor/src/components/post-featured-image/index.js

```diff
@@ -174,52 +198,80 @@ function PostFeaturedImage( {
 						modalClass="editor-post-featured-image__media-modal"
 						render={ ( { open } ) => (
 							<div className="editor-post-featured-image__container">
-								<Button
-									__next40pxDefaultSize
-									ref={ toggleRef }
-									className={
-										! featuredImageId
-											? 'editor-post-featured-image__toggle'
-											: 'editor-post-featured-image__preview'
-									}
-									onClick={ open }
-									aria-label={
-										! featuredImageId
-											? null
-											: __(
-													'Edit or replace the featured image'
-											  )
-									}
-									aria-describedby={
-										! featuredImageId
-											? null
-											: `editor-post-featured-image-${ featuredImageId }-describedby`
-									}
-									aria-haspopup="dialog"
-									disabled={ isLoading }
-									accessibleWhenDisabled
-								>
-									{ !! featuredImageId && media && (
-										<img
-											className="editor-post-featured-image__preview-image"
-											src={ mediaSourceUrl }
-											alt={ getImageDescription( media ) }
-										/>
-									) }
-									{ isLoading && <Spinner /> }
-									{ ! featuredImageId &&
-										! isLoading &&
-										( postType?.labels
-											?.set_featured_image ||
-											DEFAULT_SET_FEATURE_IMAGE_LABEL ) }
-								</Button>
+								{ noImageData ? (
+									<p>
+										{ __(
+											'Could not retrieve the featured image data.'
+										) }
+									</p>
+								) : (
+									<Button
+										__next40pxDefaultSize
+										ref={ toggleRef }
+										className={
+											! featuredImageId
+												? 'editor-post-featured-image__toggle'
+												: 'editor-post-featured-image__preview'
+										}
+										onClick={ open }
+										aria-label={
+											! featuredImageId
+												? null
+												: __(
+														'Edit or replace the featured image'
+												  )
+										}
+										aria-describedby={
+											! featuredImageId
+												? null
+												: `editor-post-featured-image-${ featuredImageId }-describedby`
+										}
+										aria-haspopup="dialog"
+										disabled={ isLoading }
+										accessibleWhenDisabled
+									>
+										{ !! featuredImageId && media && (
+											<img
+												className="editor-post-featured-image__preview-image"
+												src={ mediaSourceUrl }
+												alt={ getImageDescription(
+													media
+												) }
+											/>
+										) }
+										{ ( isLoading ||
+											isRequestingFeaturedImageMedia ) && (
+											<Spinner />
+										) }
```

Comment:
Do we need to add the `isRequestingFeaturedImageMedia` condition here? The values selected via `hasFinishedResolution` rarely work for loading states.

Reply:
Actually, this is the bit that makes the Spinner work also when normally selecting an image from the Media library. Previously, the spinner appeared only _when dragging_ an image to the drop zone. The existing `isLoading` name is a bit misleading. At first I thought it was to check the actual 'loading' of an image but no, it is true only when dragging. A better name would have been `isUploading` or something like that.

Reply:
The block editor only displays spinners when uploading an image; We shouldn't display a snipper when image data is fetched from the server.

P.S. Snippers should always be a last resort; they make sense as uploading indicators but should be avoided as loading indicators whenever possible, IMO>

Reply:
> The block editor only displays spinners when uploading an image

Hum, I'm not sure that would be a good design principle. Not even sure it's a documented guideline? I see Spinners used for many things e.g. loading Pages, Patterns, etc. also Spinners used in documentation examples for other things like delete buttons etc.
Anyways, without the Spinner, this UI would only show an _empty_ button which isn't a good pattern IMHO. Screenshot:

![Screenshot 2024-11-14 at 09 32 30](https://github.com/user-attachments/assets/89874dfc-6726-4b46-addc-6c0e2ef87e66)